src/util/safe_pointer.h

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************\
```

Comment:
It would be good if this was interface-compatible with unique_ptr, etc, e.g. use constructors and overload * and -> for convenient access